src/components/Button/TwoStateButton/TwoStateButton.tsx

```diff
@@ -177,6 +180,36 @@ export const TwoStateButton: FC<TwoStateButtonProps> = React.forwardRef(
       return iconSize;
     };
 
+    const getLoaderSize = (): LoaderSize => {
+      let loaderSize: LoaderSize;
+      if (size === ButtonSize.Flex && largeScreenActive) {
+        loaderSize = LoaderSize.Small;
+      } else if (
+        size === ButtonSize.Flex &&
+        (mediumScreenActive || smallScreenActive)
+      ) {
+        loaderSize = LoaderSize.Medium;
+      } else if (size === ButtonSize.Flex && xSmallScreenActive) {
+        loaderSize = LoaderSize.Large;
+      } else if (size === ButtonSize.Large) {
+        loaderSize = LoaderSize.Large;
+      } else if (size === ButtonSize.Medium) {
+        loaderSize = LoaderSize.Medium;
+      } else if (size === ButtonSize.Small) {
+        loaderSize = LoaderSize.Small;
+      }
+      return loaderSize;
+    };
+
+    const getButtonLoader = (): JSX.Element =>
+      loading && (
+        <Loader
+          classNames={styles.loader}
+          dotClassNames={styles.loaderDot}
```

Comment:
best i can tell from poking around in dev tools, you might be able to avoid the overhead of explicitly setting the color for each implementation by setting `background: currentColor`, and then it will match the text color of the given button

Reply:
Unfortunately, the bg color of loader is the bg color of button. In this case it would not work.

Reply:
There are loader implementations outside of button in vscode, we should keep the explicit color definitions